lib/pushmi_pullyu/swift_deposit.rb

```diff
@@ -0,0 +1,58 @@
+require 'openstack'
+
+class PushmiPullyu::SwiftDepositer
+
+  def initialize(connection: nil, container: nil)
+    raise 'conection can not be nil' if connection.nil?
+    raise 'container can not be nil' if container.nil?
+
+    user         = connection[:username]
+    pass         = connection[:password]
+    tenant       = connection[:tenant]
+    endpoint     = connection[:endpoint]
+    service_type  = 'object-store'
+    auth_method   = 'password'
+    auth_version = connection[:auth_version]
+
+    @swift_connection = OpenStack::Connection.create(username: user,
+                                                     api_key: pass,
+                                                     auth_method: auth_method,
+                                                     auth_url: "#{endpoint}/auth/#{auth_version}",
+                                                     authtenant_name: tenant,
+                                                     service_type: service_type)
+    @swift_container = container
+    @logger = PushmiPullyu.logger
+  end
+
+  def deposit_file(file_name)
+    # check if file exists
+    raise "File #{file_name} does not exist" unless File.file?(file_name)
+    file_base_name = File.basename(file_name)
+
+    # calculate file hash
+    hash = Digest::MD5.file(file_name).hexdigest
+
+    # check that container exits
+    raise "Container #{@swift_container} does not exist" unless @swift_connection.container_exists?(@swift_container)
```

Comment:
Maybe just call `era_container = @swift_connection.container(@swift_container)` or something and check if era_container is an error/nil (exception is raised etc)? Otherwise your doing two API calls here. One for checking if it exists and another for actually going and grabbing the container.